### DISCUSSION TITLE
Banana pi bpi-r4 on sale now

### DISCUSSION THREAD

**Discussion by TimOfAllTrades** [fakemanhk/openwrt-jp-ipoe#25] [General]
I have had my eye on this router for a while now.  It seems to be pretty solid in terms of specs.  I saw it on Ali Express for 16,000 yen.  Anyone else have an opinion on this?

Still can't help but wonder if it is possible to plug the optical fiber into an SFP port and ditch the NTT ONU entirely.

**Comment by fakemanhk** [comment]
I already have this board about 1 month ago, was a bit busy that I haven't started to play around with it yet (only compiled the testing firmware). But to use the optics directly you need a GPON (for 1G) or XG-PON (10G), while I am not sure if the GPON can be used on SFP+ slot, also the module needs to be configured, not sure if there's any known config (I heard that we might request NTT to send a GPON/XG-PON but not sure)

If you are on 1G plan there is existing Banana Pi BPI-R3 which has SFP slot and well supported by OpenWrt as well.

**Comment by TimOfAllTrades** [comment]
I also thought about the R3, although the 2GB DDR memory leaves a bit to be desired.  But there is a really nice aluminum case for the R3 that I haven't seen for the R4.

**Reply by fakemanhk** [reply]
As a router 2GB memory is more than enough, my NanoPi R2S has 1G memory and I am already able to put LXC with PiHole together, so 2GB memory can do lots of things.
In case you really want more memory with 1G connectivity and you don't mind to continuing to  use ISP ONU, you can get NanoPi R4S (4GB, non-SE version), I also have this one and the performance is actually great.

**Reply by fakemanhk** [reply]
Just forgot to mention, another pretty good one can be used: Buffalo WXR-5950AX12/WXR-6000AX12, I bought a used one with 20K JPY which is not bad.

**Reply by TimOfAllTrades** [reply]
Thanks for the suggestions.  I think I would be pretty much set on either the R4 or R3 (leaning towards R4 as soon as a decent case comes out for it).

I found this youtube video of some guy who seemingly has the SFP transceiver ONU for a modem/router combo device.
https://www.youtube.com/watch?v=X-X8A6Y8Big

A bit more digging around I found this NTT page which seems that an SFP ONU is available (at least for businesses).  
https://business.ntt-east.co.jp/service/onu/

I assume that if I somehow got that SFP transceiver from NTT, it should just be plug in and it will work as is?

Thanks for your help btw.

**Comment by missing233** [comment]
Only FLET'S HIKARI NEXT (NGN) provides SFP-ONU. 
Both NTT and KDDI use EPON, and there is an 802.1x EAP authentication present, so it's not possible to use third-party SFP-ONUs. 
In Japan, only NURO Hikari uses GPON, which allows for the easy replacement of the ISP ONT with gpon sticks like the MA5671A.
